### PR TITLE
Separates syntax and security checks

### PR DESCRIPTION
**Pull request by JPrevost**
Why are these changes being introduced:

* Running the syntax check after all plugins are installed is really slow and since we don't maintain that code any issues won't be easily addressable by us

Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/LM-73

How does this address that need:

* creates separate composer tasks for security and syntax
* runs the syntax job before installing plugins

### Secrets

- [x] All new secrets have been added to Pantheon tiers
- [x] Relevant secrets have been updated in Github Actions
- [x] All new secrets documented in README

### Documentation

- [ ] Project documentation has been updated
- [x] No documentation changes are needed

### Accessibility

- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)

### Stakeholder approval

- [ ] Stakeholder approval has been confirmed
- [x] Stakeholder approval is not needed

### Dependencies

NO dependencies are updated

## Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] The changes have been verified
- [x] The documentation has been updated or is unnecessary
- [x] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,11 +28,14 @@ jobs:
...
+      run: composer syntax
+
     - name: Check Composer lock file is up to date
       run: composer validate --no-check-all
 
     - name: Install Composer dependencies
       run: composer install --no-progress --prefer-dist --optimize-autoloader
 
     - name: Run tests
-      run: composer test
```
Comment (matt-bernhardt) [inline .github/workflows/ci.yml:38]: Could you please update https://github.com/MITLibraries/mitlib-wp-network/blob/master/.github/README.md?plain=1#L140-L144 with this new setup?

File: composer.json [hunk 2]
```diff
@@ -31,7 +31,13 @@
     }
   ],
   "keywords": [
-    "bedrock", "composer", "roots", "wordpress", "wp", "wp-config", "pantheon"
```
Comment (matt-bernhardt) [inline composer.json:34]: I'm assuming that these changes are something a linter did - they seem to be an automated "put each value on its own line" sort of thing, which makes sense to me.

**Review by matt-bernhardt — CHANGES_REQUESTED** [review]
I'm curious how we are going to find this new workflow, but am definitely willing to try it. The decreased time is significant (it looks like this drops the CI time to ~20 seconds from several minutes).

My only request is to update the `.github/readme.md` file where there's a reference to `composer test` that should now be `composer security` (and if changed words around that make sense, that's fine too)

File: composer.json (hunk 2 above)
Comment (JPrevost) [reply composer.json:34]: Yes, I should have noted that in side-effects. I couldn't get my editor to _not_ reformat the json lol

**Review by matt-bernhardt — APPROVED** [review]
Thanks for this! I'm looking forward to having this turnaround be faster.

:shipit:

**Pull request merged**